daffodil-test/src/test/resources/org/apache/daffodil/section00/general/generalSchedmaDefinitionError.tdml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<tdml:testSuite
+  xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:daf="urn:ogf:dfdl:2013:imp:daffodil.apache.org:2018:ext"
+  xmlns:dfdlx="http://www.ogf.org/dfdl/dfdl-1.0/extensions"
+  xmlns:ex="http://example.com"
+  xmlns:tns="http://example.com">
+
+  <tdml:defineSchema name="s1">
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat" dfdl:lengthKind="delimited"/>
+    <xs:element name="root" type="xs:string"/>
+  </tdml:defineSchema>
+
+<!--
+    Test Name: schema_definition_err
+       Schema: s1
+         Root: root
+      Purpose:
+      Daffodil should issue a specific diagnostic whenever it finds dfdl: prefixed attributes in the annotation elements dfdl:format, dfdl:element, dfdl:simpleType, dfdl:sequence, dfdl:choice, etc.
+      It should point out specifically that the author probably wanted to remove the prefix dfdl:
+-->
+
+  <tdml:parserTestCase name="schema_definition_err" root="root"
+    model="s1"
+    description="">
+    <tdml:document />
+ <tdml:errors>
+<tdml:error>SAXParseException</tdml:error>
+ <tdml:error> Attribute 'dfdl:lengthKind' is not allowed to appear in element 'dfdl:format'.</tdml:error>
```

Comment:
We usually try to make expected error message somewhat generic so that if we ever change the error message we don't need to update the tests. Especial since this error message comes from SAX, which we don't have any control over. It's also worth looking for the word "Schema Definition Error" since that's the kind of error that we should expect to see.

So I would suggest this wants to be something like
```xml
<tdml:error>Schema Definition Error</tdml:error>
<tdml:error>dfdl:lengthKind</tdml:error>
<tdml:error>dfdl:format</tdml:error>
```
Maybe that's too generic, but the idea is that we just look for certain keywords/phrases that must exist in the error, rather than looking for the entire error message. I think it's also worth loo